modules/arrow/src/schema/geoarrow-utils.ts

```diff
@@ -0,0 +1,522 @@
+// loaders.gl, MIT license
+// Copyright (c) vis.gl contributors
+
+import {Data, Vector, Schema, Table} from 'apache-arrow';
+import {BinaryFeatureCollection as BinaryFeatures} from '@loaders.gl/schema';
+import {
+  Feature,
+  MultiPolygon,
+  Position,
+  Polygon,
+  MultiPoint,
+  Point,
+  MultiLineString,
+  LineString
+} from 'geojson';
+
+type RawArrowFeature = {
+  encoding?: string;
+  data: any;
+};
+
+export enum GEOARROW_ENCODINGS {
+  MULTI_POLYGON = 'geoarrow.multipolygon',
+  POLYGON = 'geoarrow.polygon',
+  MULTI_LINESTRING = 'geoarrow.multilinestring',
+  LINESTRING = 'geoarrow.linestring',
+  MULTI_POINT = 'geoarrow.multipoint',
+  POINT = 'geoarrow.point',
+  WKB = 'geoarrow.wkb',
+  WKT = 'geoarrow.wkt'
+}
+
+const GEOARROW_COLUMN_METADATA_KEY = 'ARROW:extension:name';
+
+
+// get geometry columns from arrow table
+export function getGeometryColumnsFromArrowTable(arrowTable: Table): Map<string, {encoding: string}> {  
+  const geometryColumns = new Map<string, any>();
+  const schema = arrowTable.schema;
+  for (let i = 0; i < schema.fields.length; i++) {
+    const field = schema.fields[i];
+    const metadata = field.metadata;
+    if (metadata && metadata.has(GEOARROW_COLUMN_METADATA_KEY)) {
+      const geoEncoding = metadata.get(GEOARROW_COLUMN_METADATA_KEY);
+      geometryColumns.set(field.name, {encoding: geoEncoding});
+    }
+  }
+  return geometryColumns;
+}
+
+/**
+ * get geoarrow encoding from geoarrow column
+ */
+export function getGeoArrowEncoding(
+  schema: Schema,
+  geometryColumnName: string 
+): string {
+  const field = schema.fields.find(field => field.name === geometryColumnName);
+  return field?.metadata?.get(GEOARROW_COLUMN_METADATA_KEY) || '';
+}
+
+/**
+ *  update bounds from geoarrow samples
+ *
+ * @param flatCoords the flattend coordinates array from one chunk of geoarrow column
+ * @param nDim the number of dimensions of the coordinates
+ * @param bounds the bounds to be updated
+ * @param sampleSize how many samples to be used to update the bounds, default is 1000 per chunk
+ */
+function updateBoundsFromGeoArrowSamples(
+  flatCoords: Float64Array,
+  nDim: number,
+  bounds: [number, number, number, number],
+  sampleSize: number = 100
+): void {
+  const numberOfFeatures = flatCoords.length / nDim;
+  const sampleStep = Math.max(Math.floor(numberOfFeatures / sampleSize), 1);
+
+  for (let i = 0; i < numberOfFeatures; i += sampleStep) {
+    const lng = flatCoords[i * nDim];
+    const lat = flatCoords[i * nDim + 1];
+    if (lng < bounds[0]) {
+      bounds[0] = lng;
+    }
+    if (lat < bounds[1]) {
+      bounds[1] = lat;
+    }
+    if (lng > bounds[2]) {
+      bounds[2] = lng;
+    }
+    if (lat > bounds[3]) {
+      bounds[3] = lat;
+    }
+  }
+}
```

Comment:
I haven't yet but plan to create a repo like `geoarrow-js` to consolidate some operations like this. I implemented this "totalBounds" operation directly on an Arrow JS Vector here: https://github.com/geoarrow/deck.gl-layers/blob/8449634e9403f8159b987264fa35b1085affbbc0/src/geoarrow/bbox.ts